helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/ResourceAssignmentOptimizerAccessor.java

```diff
@@ -316,6 +295,32 @@ private AssignmentResult updateAssignmentFormat(InputFields inputFields,
     return idealStateFormatResult;
   }
 
+  /*
+  * Return if there are instance exists in more than one lists in InstanceChangeMap
+  */
+  private void validateNoIntxnInstanceChange(InputFields inputFields) {
+    Set<String> tempSet = new HashSet<>();
+    List<Collection<String>> inputs = new ArrayList<>();
```

Comment:
Why we need this list? Adding and checking with the tempSet only should be enough?

Reply:
Should be enough. 
What I have in mind is that, when we have multiple list to check for intersection, I do not want to add all the list to the hashSet. The largest one could be read only. It would safe some space....

Reply:
I see. My preference is simpler code. But it's not a strong preference. Your call.

Reply:
But on my second thought, the sorting will be expensive comparing with the HashSet addition, no?